src/pages/DelegatoryValidator/StakeDialog.tsx

```diff
@@ -1,51 +1,134 @@
 };
 
 export default function StakeDialog({
   handleDialogClose,
   isDialogOpen,
+  accountResource,
 }: StakeDialogProps) {
+  const lockedUntilSecs = accountResource
+    ? BigInt((accountResource.data as any).locked_until_secs)
+    : null;
+  const [stakedAmount, setStakedAmount] = useState<string>();
+  const [stakedAmountError, setStakedAmountError] = useState<boolean>(false);
+  const handleStakedAmountChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const amt = event.target.value;
+    setStakedAmountError(isNaN(Number(amt)));
+    setStakedAmount(amt);
+  };
+
   return (
-    <Dialog open={isDialogOpen} onClose={handleDialogClose} fullWidth={true}>
-      <DialogTitle>Stake into the pool</DialogTitle>
-      <DialogContent>
-        <FormHelperText>Enter amount</FormHelperText>
-        <OutlinedInput
-          placeholder="0"
-          endAdornment={<InputAdornment position="end">APT</InputAdornment>}
-        />
-      </DialogContent>
-      <ContentBox>
-        <ContentRow title="Operator commission" value={null} />
-        <ContentRow title="Compound rewards" value={null} />
-        <ContentRow title="Next unlock in" value={null} />
-      </ContentBox>
-      <ContentBox>
-        <ContentRow title="Transaction fee" value={null} />
-      </ContentBox>
-      <DialogActions>
-        <Button onClick={handleDialogClose}>Deposit</Button>
-      </DialogActions>
-      <DialogContent>
-        Be aware that you will be able to see your funds in the pool after 1
-        epoch (~2 hours) due to the delay time
-      </DialogContent>
+    <Dialog open={isDialogOpen} onClose={handleDialogClose}>
+      <Box sx={{marginX: 4}}>
+        <DialogContent>
+          <IconButton
+            onClick={handleDialogClose}
+            sx={{
+              position: "absolute",
+              right: 12,
+              top: 12,
+              color: (theme) => theme.palette.grey[500],
+            }}
+          >
+            <CloseIcon />
+          </IconButton>
+        </DialogContent>
+        <DialogTitle>
+          <Typography variant="h5" textAlign="center">
+            Stake Into The Pool
+          </Typography>
+        </DialogTitle>
+        <DialogContent>
+          <FormHelperText
+            sx={{
+              fontSize: "1rem",
+              lineHeight: "1.1rem",
+            }}
+          >
+            Enter amount
+          </FormHelperText>
+          <OutlinedInput
+            onChange={handleStakedAmountChange}
+            value={stakedAmount}
+            sx={{marginTop: 2}}
+            fullWidth
+            placeholder="0"
+            endAdornment={
+              <InputAdornment
+                position="end"
+                sx={{ml: 0.5, marginTop: "0!important"}}
+              >
+                APT
+              </InputAdornment>
+            }
+          />
+          {stakedAmountError && <FormHelperText error>Error</FormHelperText>}
+        </DialogContent>
```

Comment:
not completely sure how MUI Dialog works, but I would assume it should have a  one `<DialogContent >` component..?

```
<DialogTitle />
   <DialogContent >
      ... content ....
   </DialogContent >
<DialogActions />
```

Reply:
you were referring to have this instead?
```
<DialogContent >
      ... single content ....
   </DialogContent >
<DialogContent >
      ... single content ....
   </DialogContent >
```
the built-in css layouts for `DialogContent` is a bit too much padding/margin for us, that's why i keep all the content into a single `DialogContent`

Reply:
looks like there is 4 different `DialogContent` components inside the parent `Dialog` component, is that how it should be..?

Reply:
good call. by looking at the MUI site: https://mui.com/material-ui/react-dialog/, `DialogContent` is used to wrap about the content. will remove some of them and use `Stack` for spacing around each block.